specifyweb/context/views.py

```diff
@@ -258,6 +258,10 @@ def user(request):
     data = obj_to_data(request.specify_user)
     data['isauthenticated'] = request.user.is_authenticated
     data['available_collections'] = users_collections(connection.cursor(), request.specify_user.id)
+
+    if not hasattr(request.specify_user_agent, '_meta'):
+        return HttpResponseBadRequest('user does not have an agent')
+
     data['agent'] = obj_to_data(request.specify_user_agent)
```

Comment:
Better yet just set `data['agent']` to None if `request.specify_user_agent` is None. Then the request can succeed.

Reply:
Fixed